templates/template.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+require "erb"
+require "fileutils"
+require "yaml"
+
+module RBS
+  class Template
+    class Field
+      attr_reader :name
+      attr_reader :c_name
+      attr_reader :c_type
+
+      def initialize(yaml)
+        @name = yaml["name"]
+        @c_name = yaml["c_name"] || @name
+        @c_type = "VALUE"
+      end
+    end
+
+    class Type
+      attr_reader :full_name
+      attr_reader :name
+      attr_reader :c_function_name
+      attr_reader :c_constant_name
+      attr_reader :parent_c_constant_name
+      attr_reader :fields
+
+      def initialize(yaml)
+        @full_name = yaml["name"]
+        @name = yaml["name"].split("::").last
+        @c_function_name = if @full_name =~ /^RBS::Types/
+          name = @full_name.gsub("::", "_").underscore
+          name.gsub("_types_", "_") # TODO: should we rename these methods?
+        else
+          name = @full_name.gsub("::", "_").underscore
+          name.gsub("_declarations_", "_decl_") # TODO: should we rename these methods?
+        end
+        @c_constant_name = @full_name.gsub("::", "_")
+        @fields = yaml["fields"].map { |field| Field.new(field) }
+        @parent_c_constant_name = @full_name.split("::")[0..-2].join("::").gsub("::", "_")
```

Comment:
I wonder if we could be smarter and compose the `config.yml` nodes list using some kind of nesting?

Reply:
We could, but I think it would complicate the structure that would make other things more complex (like automated parsing with `jq`, which i've used a few times)